### PR TITLE
4.2 final fixes

### PR DESCRIPTION
**Pull request by ryancarlson**
Add package code and language code to emitted button tap event …
- this will allow analytics to scope the event properly 
- this can actually be useful in many other ways

Also revert to a single tap to show/hide nav bar.. multiple was confusing.

### CODE REVIEW

**Comment by ryancarlson** [comment]
@michaelharro I'm going to move ahead with this so I can get things submitted to Apple, but feedback is welcome and i can resubmit if needed.

File: GTViewController/Classes/Nav Classes/GTFollowupViewController.m [hunk 1]
```diff
@@ -28,6 +28,9 @@ @interface GTFollowupViewController ()
 @property (weak, nonatomic)         UITextField *activeField;
 @property (assign, nonatomic)       CGFloat originalHeight;
 @property (assign, nonatomic)       CGRect visibleFrame;
+@property (strong, nonatomic)       NSString *packageCode;
+@property (strong, nonatomic)       NSString *LanguageCode;
```
Comment (michaelharro) [inline GTViewController/Classes/Nav Classes/GTFollowupViewController.m:32]: Should be lowercase l

File: GTViewController/Classes/Nav Classes/GTFollowupViewController.m [hunk 2]
```diff
@@ -321,6 +329,16 @@ - (BOOL)validateFields {
     return YES;
 }
 
+- (NSString *)currentLanguageCode {
```
Comment (michaelharro) [inline GTViewController/Classes/Nav Classes/GTFollowupViewController.m:332]: The property statement already generates getters and setters for you. If you want to access it as currentLanguageCode I would update your property statement

File: GTViewController/Classes/Nav Classes/GTFollowupViewController.m [hunk 3]
```diff
@@ -101,6 +104,11 @@ - (void)setFollowupView:(GTFollowupModalView *)followupModalView andThankYouView
     return;
 }
 
+- (void)setPackageCode:(NSString *)packageCode andLanguageCode:(NSString *)languageCode {
```
Comment (michaelharro) [inline GTViewController/Classes/Nav Classes/GTFollowupViewController.m:107]: The and in this method name is nonstandard but still very readable. I would leave it but for future methods try using the standard of no and

File: GTViewController/Classes/Nav Classes/GTViewController.m [hunk 4]
```diff
@@ -1190,6 +1193,16 @@ - (void)dismissFollowupModal {
     
 }
 
+
+- (NSString *)currentPackageCode {
```
Comment (michaelharro) [inline GTViewController/Classes/Nav Classes/GTViewController.m:1197]: Please use property statements to generate your setters and getters unless you really need to do something custom

File: GTViewController/Classes/Parsing Classes/GTPage.m [hunk 5]
```diff
@@ -572,6 +572,20 @@ - (void)didReceiveTapOnButton:(UISnuffleButton *)button {
 	
 }
 
+- (NSString *)currentPackageCode {
```
Comment (michaelharro) [inline GTViewController/Classes/Parsing Classes/GTPage.m:575]: This is an appropriate use of a custom getter

**Pull request merged**